projects/step-core/src/lib/directives/clamp-fade.directive.ts

```diff
@@ -0,0 +1,147 @@
+import {
+  AfterViewInit,
+  Directive,
+  ElementRef,
+  HostBinding,
+  NgZone,
+  OnDestroy,
+  Renderer2,
+  input,
+  effect,
+} from '@angular/core';
+
+const FADE_HEIGHT_PX = 24;
+
+@Directive({
+  selector: '[stepClampFade]',
+  standalone: true,
+})
+export class ClampFadeDirective implements AfterViewInit, OnDestroy {
+  lines = input(5);
+  fadeBackground = input<string | undefined>();
+
+  @HostBinding('attr.data-step-clamp-fade') private attribute = '';
+
+  private resizeObserver?: ResizeObserver;
+  private mutationObserver?: MutationObserver;
+  private overlayElement: HTMLElement | null = null;
+  private supportsMaskImage =
+    typeof CSS !== 'undefined' &&
+    (CSS.supports?.('mask-image', 'linear-gradient(black, transparent)') ||
+      CSS.supports?.('-webkit-mask-image', 'linear-gradient(black, transparent)'));
+
+  private isClamped = false;
+
+  constructor(
+    private elementReference: ElementRef<HTMLElement>,
+    private renderer: Renderer2,
+    private zone: NgZone,
+  ) {
+    effect(() => {
+      const value = String(this.lines());
+      this.renderer.setStyle(this.elementReference.nativeElement, '-webkit-line-clamp', value);
+      queueMicrotask(() => this.updateClampedState());
+    });
+  }
```

Comment:
1. According to our coding guide, we're using `inject` function instead of constructor. Effect can be defined as field in this case
2. Also according to our coding guide, all injections should be prefixed with underscore
```
private _elementReference = inject<ElementRef<HtmlElement>>(ElementRef);
private _renderer = inject(Renderer2);
private _zone = inject(NgZone);

private effectUpdateClampState = effect(() => {
   ...
});
```